### PR TITLE
Bunch of individual changes, alongside pulling in some new sprites

### LINKED ISSUE #254: Windoors do not play an opening animation
dreamseeker_itjTBy0Dkk

They just snap to the open state instantly. When closing they snap to the closed state instantly for a single frame, then play the animation properly. Still impassible during the invisible animation.

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

Just a ton of small changes I've had cooking for the past few weeks. Contains sprites from Draco and Meyhazah.
I've also taking the liberty of fully implementing some sprites added recently, and have resurrected viro's sign sprites.

[Adds 4 corner style airlocks (centcom and maint) alongside morge sprites (draco)](https://github.com/wall-nerds/wallening/commit/6181256832b19c5fa294650589d0f42edcc21bf9)

[Adds construction sprites to gags, adds support for inheriting airlock stuff to assemblies](https://github.com/wall-nerds/wallening/commit/e12b9281eb9fc90489e2dc727120c90e06bf7ae8)

[Cleans up greyscale application, adds unit test](https://github.com/wall-nerds/wallening/commit/dc5f714e1a711c2a523871138a15030da12ae1c0)

I want to test for airlock icon states, so I don't miss shit or anythin

[adds a construction state to windowed airlocks](https://github.com/wall-nerds/wallening/commit/24fa3e46fd5efe5c6ccb581fffca2243ed2d3d1e)

[Adds construction states to a bunch of airlocks, fixes a few small issues](https://github.com/wall-nerds/wallening/commit/5f82d2dbb0cac41d8602d2c03f87cf93072a6b73)

[Readds viro's sign sprites](https://github.com/wall-nerds/wallening/commit/53dd817e607b1ca3f23d70bcea7d8476f93f25fa)

#82 

This change got dropped YEARS ago because I merged it into experimental incorrectly and then never pulled that over to stable. I am a fool

I've added all the missing sprites in the same style, so we should be all good now

[Moves material airlocks to the new system](https://github.com/wall-nerds/wallening/commit/9cac853a2619b2b585b98d8a3e63c92de05bc025)

Trivial, we just give them the default config and extend their colors to match the length required.

Also resolves a few bugs with the airlock typepath handling stuff

[adds table buttons (meyhaz)](https://github.com/wall-nerds/wallening/commit/474104884264f78cebf726bdbb670ccb4f910e12)

I've added a bunch of subtypes to make this work, please advise map people.

[Cleans up a few airlocks](https://github.com/wall-nerds/wallening/commit/f14d06b8e52d1868342794500b52c075b0993a23)

Highsec and vault had a trim on the directionals that didn't quite work

Wood was shifted right one on opened which lead to a weird jumping effect

Shuttle had its center inconsistent, and had a fill improperly inserted in some of the animations.

Mourge had a 0.5 second delay on the start of opening which seemed... odd

[Implements new shutters (draco)](https://github.com/wall-nerds/wallening/commit/7b0b9dbcb92dd56a7e9f431f063dccac8e005ce9)

Adds nuclear and seethrough shutters

Implements emissives for shutters so they look nice in the dark

[Fills out most all of the remaining material doors (meyhaz)](https://github.com/wall-nerds/wallening/commit/2d5708d9e86af8d39c7f8b04daa11c75088e1f5e)

Still missing resin, we'll see what baron's alien walls look like.

[Implements proper delays for the mourge airlock, takes animation delay variation into account for wait times](https://github.com/wall-nerds/wallening/commit/6a6006189a78b105116d3bd9f608173053bcbf25)

[Removes the greyscale test airlocks (testing DONE!)](https://github.com/wall-nerds/wallening/commit/a1aa8e4c6906313c76aa5f8f61dbd45b6864ffac)

[Adds cult runed airlock closing animation (draco)](https://github.com/wall-nerds/wallening/commit/2b5c50f50645036c9ade8d22bc7c6a05d95077e9)

[Removes edging from puzzledoors, converts then to open_top/bottom](https://github.com/wall-nerds/wallening/commit/42b0b56d9c49680e9eb7ffa64daaa7050992d012)

[Converts mineral doors to open_top/bottom](https://github.com/wall-nerds/wallening/commit/e6263badbda368691172c21a8f399c15b68025e6)

[Implements variable open delays, expands implementations](https://github.com/wall-nerds/wallening/pull/237/commits/c8ce3c2b0144f8bbeb19b70bc894d70f14c87356)
[c8ce3c2](https://github.com/wall-nerds/wallening/pull/237/commits/c8ce3c2b0144f8bbeb19b70bc894d70f14c87356)

Airlock opening delay should vary with how long the animation actually takes (since people are playing with them now). This does mean if you give me a 3 second opening animation that fucker will take 3 seconds to open.

Implements that here and for doors. Adds delay handling to the hatches, vault/highsec and cult doors.

[updated halon/nitrium sprites from our good pal arcane](https://github.com/wall-nerds/wallening/pull/237/commits/264d8bac1eff3b11c2f7062747f8952894b7cf91)

[prevents jumping in windoor animation](https://github.com/wall-nerds/wallening/pull/237/commits/67fe05b89f14bcdbd80494d52bbbb1f8f908fe31)

Closes #254

### CODE REVIEW

**Comment by san7890** [comment]
You asked for my thought on table buttons and I'm stuck wondering if the pattern can't be... better somehow... it's obvious we need table buttons but I just can't stop thinking about if there would be a better way to accomplish it. are you able to split that out into another PR? i don't want to hold up this one while i think on a better way to tackle it (and we can just send that PR through if I can't figure it out)

**Comment by LemonInTheDark** [comment]
> You asked for my thought on table buttons and I'm stuck wondering if the pattern can't be... better somehow... it's obvious we need table buttons but I just can't stop thinking about if there would be a better way to accomplish it. are you able to split that out into another PR? i don't want to hold up this one while i think on a better way to tackle it (and we can just send that PR through if I can't figure it out)

I absolutely could if that's something you want. I am still working on stuff here that's fit with this pr though so it's no rush regardless.

**Comment by san7890** [comment]
splitting it out would just be nicer for scrolling through it repeatedly and dwelling on potential solutions while i think about it (less clutter with other changes) but it's up to whatever time you have

**Comment by LemonInTheDark** [comment]
I am a bit busy atm, but if that's what you're after I raise you https://github.com/wall-nerds/wallening/pull/237/commits/474104884264f78cebf726bdbb670ccb4f910e12

**Comment by san7890** [comment]
anyways my thoughts on directionals are it should be `/obj/machinery/button/thing/table` and not `table/thing` because having to scroll a massive thing of just `/table` stuff seems quite taxing on the mapping process, imagining the object tree in a map editor:

```txt
* /obj
 - /machinery
  - /button
   - /thing <- both directional walls and the table subtype are right here for this specific type!
    - /directional
    - /table 
```

It would even be less copypasta code because all we would need is `MAPPING_TABLE_BUTTON_HELPER(/obj/machinery/button/thing)` to set the icon states/dirs to make this work instead of copypasting the access and whatever, just one line. that's how i see it anyways, let me know if that's not feasible.

**Comment by LemonInTheDark** [comment]
> anyways my thoughts on directionals are it should be `/obj/machinery/button/thing/table` and not `table/thing` because having to scroll a massive thing of just `/table` stuff seems quite taxing on the mapping process, imagining the object tree in a map editor:
> 
> ```
> * /obj
>  - /machinery
>   - /button
>    - /thing <- both directional walls and the table subtype are right here for this specific type!
>     - /directional
>     - /table 
> ```
> 
> It would even be less copypasta code because all we would need is `MAPPING_TABLE_BUTTON_HELPER(/obj/machinery/button/thing)` to set the icon states/dirs to make this work instead of copypasting the access and whatever, just one line. that's how i see it anyways, let me know if that's not feasible.

I was concerned that sort of organization would be annoying, but I can 100% make that happen

**Comment by san7890** [comment]
it would be less annoying, see how "any" and "all" access helpers presently works.

**Comment by LemonInTheDark** [comment]
Table buttons should be better now

File: icons/obj/doors/airlocks/tall/hatch/centcom.dmi
Comment (GoldenAlpharex) [inline icons/obj/doors/airlocks/tall/hatch/centcom.dmi:1]: Is it normal that the top and bottom halves don't align vertically? The bottom is off to the left by one pixel, or the top, one to the right by one pixel. Doesn't form a perfect +

Comment (LemonInTheDark) [reply icons/obj/doors/airlocks/tall/hatch/centcom.dmi:1]: image
context

Comment (LemonInTheDark) [reply icons/obj/doors/airlocks/tall/hatch/centcom.dmi:1]: I believe so but bugging draco bout this

Comment (YesterdaysPromise) [reply icons/obj/doors/airlocks/tall/hatch/centcom.dmi:1]: They do have the same offset in the current, pre-wallening sprites, and you know, the sprites are 32x56. There's no actual middle, in fact, one can argue its more centered than ones which have 1 'middle' pixel line offset either way. Plus, works nicer with the animation ang gives it bit more of industrial-grounded vibe?

Comment (LemonInTheDark) [reply icons/obj/doors/airlocks/tall/hatch/centcom.dmi:1]: https://github.com/wall-nerds/wallening/assets/58055496/c56a68da-0be3-40dc-b080-4cc1dc7bc56f

looks fine in motion

**Pull request merged**